### DISCUSSION TITLE
Simulation of Mavic2 pro isn't holding drone on air

### DISCUSSION THREAD

**Discussion by Pallavi-Sharma12** [cyberbotics/webots#6349] [General]
Hi Team,

I've been trying to perform the simulation of Mavic2 pro on Webots using Python but during simulation the drone doesn't hold on air. It oscillates up-down all the time during simulation. I've even tried cyberbotic official posted code but nothing helps.

I want my Mavic2 pro to be stable at certain altitude and fly from one coordinate to another coordinate(can be hardcoded value) using Webots with Python or C.

Please help as what I'm missing.

Sample code for Mavic2 pro controller - https://github.com/Pallavi-Sharma12/DJI_Controller/blob/main/dji.py 
Webots version - 2023

**Comment by ShuffleWire** [comment]
I've tried your code, and work just fine : the drone stabilize after about 20 sec. Maybe you didn't wait enough ? Because yes a the beginning it oscillate.

**Reply by Pallavi-Sharma12** [reply]
Hi,

Thanks for checking the code.

Same code or any Mavic2 pro sample code isn't stabilizing the drone on air, during simulation I've waited for 4-5 mins. Tried coding in C or Python, webots reinstallation but still the same :(

**Reply by ShuffleWire** [reply]
EDIT : I might I've found your bug, please read my last message :) 

try this simpler world and controller : 
[mavic.zip](https://github.com/cyberbotics/webots/files/12299589/mavic.zip)
you should get that
https://github.com/cyberbotics/webots/assets/83014015/ebd35e37-6509-44c1-ae95-b81e3ddafb15

**Reply by ShuffleWire** [reply]
and if it still doesn't work try this one
[mavic_on_slider.zip](https://github.com/cyberbotics/webots/files/12299764/mavic_on_slider.zip)
(the mavic is constrainted to a vertical movement, so no perturbation of other movements)
https://github.com/cyberbotics/webots/assets/83014015/ca9ccbea-3757-44a0-85c1-454994725efe

**Reply by ShuffleWire** [reply]
I've read https://github.com/Pallavi-Sharma12/DJI_Webots/blob/main/DJI/worlds/dji.wbt
and you seems to have removed the Damping in the World info
![image](https://github.com/cyberbotics/webots/assets/83014015/ea05e249-316a-4251-867a-8f1af91c6a15)
In my testing, if I remove it, I also get a non stable behavior !
Try adding it back, and see if it improve things

**Reply by Pallavi-Sharma12** [reply]
Thank you so much @ShuffleWire . 

Yes, adding defaultDamping helped to put Mavic2 pro on stable mode.

It's 'null' by default while creating the world.

<img width="135" alt="image" src="https://github.com/cyberbotics/webots/assets/54979921/2f68fde5-6edc-485e-ab49-704135bf218b">

Thanks again :)

**Comment by ShuffleWire** [comment]
Nice to see it working !
Cheers

**Comment by subodhnarayan** [comment]
@ShuffleWire @Pallavi-Sharma12 can i get your Mavic2pro pytho running code. Actually my task is to take up drone for 1m height and then make square in above of 3m and again setdown drown at intial point.
